src/Commands/CommandFactory.cs

```diff
@@ -106,10 +108,38 @@ private void RegisterCosmosCommands()
         cosmosContainer.AddSubGroup(cosmosItem);
 
         // Register Cosmos commands
-        databases.AddCommand("list", new DatabaseListCommand(GetLogger<DatabaseListCommand>()));
+        databases.AddCommand("list", new Cosmos.DatabaseListCommand(GetLogger<Cosmos.DatabaseListCommand>()));
         cosmosContainer.AddCommand("list", new Cosmos.ContainerListCommand(GetLogger<Cosmos.ContainerListCommand>()));
         cosmosAccount.AddCommand("list", new Cosmos.AccountListCommand(GetLogger<Cosmos.AccountListCommand>()));
-        cosmosItem.AddCommand("query", new ItemQueryCommand(GetLogger<ItemQueryCommand>()));
+        cosmosItem.AddCommand("query", new Cosmos.ItemQueryCommand(GetLogger<Cosmos.ItemQueryCommand>()));
+    }
+
+    private void RegisterDataExplorerCommands()
+    {
+        // Create Data Explorer command group
+        var dataExplorer = new CommandGroup("data-explorer", "Azure Data Explorer (Kusto) operations - Commands for managing and querying Azure Data Explorer clusters.");
+        _rootGroup.AddSubGroup(dataExplorer);
+
+        // Create Data Explorer cluster subgroups
+        var clusters = new CommandGroup("cluster", "Data Explorer cluster operations - Commands for listing clusters in your Azure subscription.");
+        dataExplorer.AddSubGroup(clusters);
+
+        var databases = new CommandGroup("database", "Data Explorer database operations - Commands for listing databases in a cluster.");
+        dataExplorer.AddSubGroup(databases);
+
+        var tables = new CommandGroup("table", "Data Explorer table operations - Commands for listing tables in a database.");
+        dataExplorer.AddSubGroup(tables);
+
+        dataExplorer.AddCommand("sample", new DataExplorer.SampleCommand(GetLogger<DataExplorer.SampleCommand>()));
+        dataExplorer.AddCommand("query", new DataExplorer.QueryCommand(GetLogger<DataExplorer.QueryCommand>()));
```

Comment:
technically, query is always scoped to a cluster. are you doing this because of the cluster argument ambiguity?

Reply:
I thought it was redundant since the cluster-uri or cluster-name are required. Please suggest a change.